src/Microsoft.VisualStudio.Editors/Common/DTEUtils.vb

```diff
@@ -113,9 +113,15 @@ Namespace Microsoft.VisualStudio.Editors.Common
         ''' </summary>
         ''' <param name="PropertyName">The name of the property to retrieve.</param>
         Public Shared Function GetProjectItemProperty(ProjectItem As ProjectItem, PropertyName As String) As [Property]
-            If ProjectItem.Properties Is Nothing Then
+            Try
+                If ProjectItem.Properties Is Nothing Then
+                    Return Nothing
+                End If
+            Catch ex As KeyNotFoundException
+                ' The issue is a limitation in CPS. The main cause is that CPS, in some cases,
+                ' it is not able to maintain the same item id for items, causing them to not be found
                 Return Nothing
-            End If
+            End Try
```

Comment:
I don't suggest we eat the exception. It is always a bad practice unless there is *no* other way. In this case, it looks like `OAProjectItem` in CPS implements this interface. It only calls `Create` in `OAProperties` so the root cause shouldn't be too complex to find. In this case, accessing a property shouldn't throw `KeyNotFoundException`. As this code expects, it should be null (Nothing) if it isn't present. Eating exceptions generally causes a lot more issues than it fixes in the long term.

Reply:
in AB#1243638 Adrian describes the CPS limitation and Dotnet ProjectSystem should handle this exception.